components/script/dom/element.rs

```diff
@@ -746,6 +746,30 @@ impl LayoutElementHelpers for LayoutJS<Element> {
                 shared_lock,
                 PropertyDeclaration::BorderRightWidth(width_value)));
         }
+
+        let cellpadding = if let Some(this) = self.downcast::<HTMLTableCellElement>() {
+            this.get_inherited_cellpadding()
+        } else {
+            None
+        };
+
+        if let Some(cellpadding) = cellpadding {
```

Comment:
I think to do this you need to add this to add another case in `Document::element_attr_will_change`... But again it should be a virtual function.